adapters/repos/db/vector/hnsw/snapshot.go

```diff
@@ -0,0 +1,108 @@
+package hnsw
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"path"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+)
+
+// PauseMaintenance makes sure that no new background processes can be started.
+// If a Combining or Condensing operation is already ongoing, the method blocks
+// until the operation has either finished or the context expired
+//
+// If a Delete-Cleanup Cycle is running (TombstoneCleanupCycle), it is aborted,
+// as it's not feasible to wait for such a cycle to complete, as it can take hours.
```

Comment:
does it mean we have to "kill" tombstone cleanup process or wait for it to finish in given timeout?

Reply:
Yes it should be killed. This should now be handled by the cycle manager